codec/dagcbor/unmarshal.go

```diff
@@ -11,7 +11,7 @@ import (
 	"github.com/polydawn/refmt/shared"
 	"github.com/polydawn/refmt/tok"
 
-	ipld "github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime"
```

Comment:
This stands out now as an anomaly (first time a non-test file in this PR's diff anyway) of including the root package rather than reaching down into the child packages and using `codec.` and `datamodel.`. It probably doesn't matter practically since it's just an indirection but maybe consistency would be good to make code more similar across packages and make refactoring easier? Maybe it'd be worth having a rule that packages in here should be explicit about pulling in named packages rather than using the root wherever possible (it's probably possible everywhere now?).

Reply:
Yes, good eye.

I haven't drawn the package graph and looked for bugs like this yet.  Could / should.

I think that, so far, as long as it doesn't fail2compile, it's harmless.  Worst case, it means this package will have an overly large transitive package import graph, as the root `ipld` package grows more imports.  But it will always be fixable as soon as we notice and are bothered by it.

It's possible that an import up to the root package like this _will_ cause a fail2compile, if it produces an import cycle.  But again, then, we just fix that immediately as soon as it appears.  I think we should generally find it clear how to do so at the time with this overall layout.

> Maybe it'd be worth having a rule that packages in here should be explicit about pulling in named packages rather than using the root wherever possible (it's probably possible everywhere now?).

Yes, exactly.  I just haven't enforced it yet :)

Reply:
This might be one of the few that _won't_ turn into a fail2compile, shortly.  One of the new features I intend to add in the root package is more single-shot helper functions that do serial->datamodel stuff... but it probably still won't reference the dagcbor package directly, outside of examples (which will exist in a `_test` package, and thus not provoke cycles).

We could add a short CI script to check that this kind of import doesn't exist, but I might punt for later down the road.

Reply:
Fixed all these.